api4/emoji.go

```diff
@@ -54,6 +48,28 @@ func createEmoji(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Allow any user with MANAGE_EMOJIS permission at Team level to manage emojis at system level
+	memberships, err := c.App.GetTeamMembersForUser(c.Session.UserId)
+
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_EMOJIS) {
```

Comment:
I don't think this outer check is necessary as the inner checks below will all cascade up to it.

Reply:
Fixed

Reply:
I realized that there is a corner case we weren't considering. If you are system admin but you aren't a team member.